src/components/TodoList.tsx

```diff
@@ -0,0 +1,66 @@
+import React from 'react';
+import { Filter } from '../types/Filter';
+import { Todo } from '../types/Todo';
+import { SingleTodo } from './SingleTodo';
+import { TempTodo } from './TempTodo';
+import { Errors } from '../types/Errors';
+
+type TodoListProps = {
+  filter: Filter;
+  todos: Todo[];
+  handleRemove: (todoId: number) => void;
+  tempTodo: Todo | null;
+  deletedTodoId: number | null;
+  deletedTodosId: number[];
+  isAddingTodo: boolean;
+  handleToggleCompleted: (id: number) => void,
+  setError: (err: Errors | null) => void,
+  onSubmit: () => void;
+};
+
+export const TodoList: React.FC<TodoListProps>
+= ({
+  filter, todos, handleRemove, setError, onSubmit,
+  tempTodo, deletedTodoId, isAddingTodo, handleToggleCompleted, deletedTodosId,
+}) => {
+  const visibleTodos = () => {
+    if (filter === 'active') {
+      return todos.filter((todo) => !todo.completed);
+    }
+
+    if (filter === 'completed') {
+      return todos.filter((todo) => todo.completed);
+    }
+
+    return todos;
+  };
+
+  return (
+    <section className="todoapp__main" data-cy="TodoList">
+      {visibleTodos().map(todo => {
```

Comment:
I think it would be great if it was variable instead of function 
```suggestion
      {visibleTodos.map(todo => {
```